cmd/internal/kube/inject.go

```diff
@@ -0,0 +1,229 @@
+package kube
+
+import (
+	"bytes"
+	"github.com/akitasoftware/akita-cli/cmd/internal/cmderr"
+	"github.com/akitasoftware/akita-cli/cmd/internal/kube/injector"
+	"github.com/akitasoftware/akita-cli/printer"
+	"github.com/akitasoftware/akita-cli/telemetry"
+	"github.com/akitasoftware/go-utils/optionals"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+)
+
+var (
+	injectFileNameFlag string
+	injectOutputFlag   string
+	projectNameFlag    string
+	secretInjectFlag   string
+)
+
+var injectCmd = &cobra.Command{
+	Use:   "inject",
+	Short: "Inject Akita into a Kubernetes deployment",
+	Long:  "Inject Akita into a Kubernetes deployment or set of deployments and output the result to stdout or a file",
+	RunE: func(_ *cobra.Command, args []string) error {
+		secretOpts := resolveSecretGenerationOptions(secretInjectFlag)
+
+		// If the user specified a secret file path, but did not specify an output file, return an error
+		if secretOpts.ShouldInject && secretOpts.Filepath.IsSome() && injectOutputFlag == "" {
+			return cmderr.AkitaErr{
+				Err: errors.New("cannot specify a secret file path without an output file (using --output or -o)"),
+			}
+		}
+
+		// Create the injector which reads from the Kubernetes YAML file specified by the user
+		injectr, err := injector.FromYAML(injectFileNameFlag)
+		if err != nil {
+			return cmderr.AkitaErr{
+				Err: errors.Wrapf(
+					err,
+					"Failed to read injection file %s",
+					injectFileNameFlag,
+				),
+			}
+		}
+
+		// Generate a secret for each namespace in the deployment if the user specified secret generation
+		secretBuf := new(bytes.Buffer)
+		if secretOpts.ShouldInject {
+			key, secret, err := cmderr.RequireAPICredentials("API credentials are required to generate secret.")
+			if err != nil {
+				return err
+			}
+
+			namespaces, err := injectr.InjectableNamespaces()
+			if err != nil {
+				return err
+			}
+
+			for _, namespace := range namespaces {
+				r, err := handleSecretGeneration(namespace, key, secret)
+				if err != nil {
+					return err
+				}
+
+				secretBuf.WriteString("---\n")
+				secretBuf.Write(r)
+			}
+		}
+
+		// Create the output buffer
+		out := new(bytes.Buffer)
+
+		// Either write the secret to a file or append it to the output
+		if secretFilePath, exists := secretOpts.Filepath.Get(); exists {
+			err = writeFile(secretBuf.Bytes(), secretFilePath)
+			if err != nil {
+				return err
+			}
+
+			printer.Infof("Kubernetes Secret generated to %s\n", secretFilePath)
+		} else {
+			// Append the secret to the output
+			out.Write(secretBuf.Bytes())
+		}
+
+		// Inject the sidecar into the input file
+		rawInjected, err := injector.ToRawYAML(injectr, createSidecar(projectNameFlag))
+		if err != nil {
+			return cmderr.AkitaErr{Err: errors.Wrap(err, "Failed to inject sidecars")}
+		}
+		// Append the injected YAML to the output
+		out.Write(rawInjected)
```

Comment:
Thanks for catching this, I'll need to update the help message. `kubectl apply` applies all resources in order since each `---` signals the start of a new Kubernetes resource/YAML file. I intentionally put the Secrets at the top of the file since any injected deployments will be dependent upon them

Reply:
I've updated the help message in [d81c238](https://github.com/akitasoftware/akita-cli/pull/206/commits/d81c2380e553609dd511498242d6f7512fa09f2d)